From: SeongJae Park <sj@kernel.org>
Date: Sun, 8 Feb 2026 12:22:20 -0800
Subject: [PATCH] mm/damon/stat: add a parameter for reading kdamond pid

It can help user-space tool, and make monitoring of DAMON's resource
consumption easier.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 7e1a9e1adcd24..055f1f69008ee 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -44,6 +44,15 @@ module_param(aggr_interval_us, ulong, 0400);
 MODULE_PARM_DESC(aggr_interval_us,
 		"Current tuned aggregation interval in microseconds");
 
+/*
+ * PID of the DAMON thread
+ *
+ * If DAMON_STAT is enabled, this becomes the PID of the worker thread.
+ * Else, -1.
+ */
+static int kdamond_pid __read_mostly = -1;
+module_param(kdamond_pid, int, 0400);
+
 static struct damon_ctx *damon_stat_context;
 
 static unsigned long damon_stat_last_refresh_jiffies;
@@ -204,6 +213,9 @@ static int damon_stat_start(void)
 	err = damon_start(&damon_stat_context, 1, true);
 	if (err)
 		return err;
+	kdamond_pid = damon_kdamond_pid(damon_stat_context);
+	if (kdamond_pid < 0)
+		return kdamond_pid;
 
 	damon_stat_last_refresh_jiffies = jiffies;
 	call_control.data = damon_stat_context;
@@ -213,6 +225,7 @@ static int damon_stat_start(void)
 static void damon_stat_stop(void)
 {
 	damon_stop(&damon_stat_context, 1);
+	kdamond_pid = -1;
 	damon_destroy_ctx(damon_stat_context);
 }
 
-- 
2.47.3

